controllers/perconaservermysqlbackup_controller.go

```diff
@@ -298,20 +300,88 @@ func getDestination(storage *apiv1alpha1.BackupStorageSpec, clusterName, creatio
 func (r *PerconaServerMySQLBackupReconciler) getBackupSource(ctx context.Context, cluster *apiv1alpha1.PerconaServerMySQL) (string, error) {
 	l := log.FromContext(ctx).WithName("getBackupSource")
 
-	orcHost := orchestrator.APIHost(cluster)
-	primary, err := orchestrator.ClusterPrimary(ctx, orcHost, cluster.ClusterHint())
+	operatorPass, err := k8s.UserPassword(ctx, r.Client, cluster, apiv1alpha1.UserOperator)
 	if err != nil {
-		return "", errors.Wrap(err, "get primary")
+		return "", errors.Wrap(err, "get operator password")
 	}
-
-	var src string
-	if len(primary.Replicas) < 1 {
-		src = primary.Key.Hostname
+	var primary string
+	var replicas []string
+	if cluster.Spec.MySQL.IsGR() {
+		fqdn := mysql.FQDN(cluster, 0)
+		db, err := replicator.NewReplicator(apiv1alpha1.UserOperator, operatorPass, fqdn, mysql.DefaultAdminPort)
+		if err != nil {
+			return "", errors.Wrapf(err, "open connection to %s", fqdn)
+		}
+		replicas, err = db.GetGroupReplicationReplicas()
+		if err != nil {
+			return "", errors.Wrap(err, "get replicas")
+		}
+		primary, err = db.GetGroupReplicationPrimary()
+		if err != nil {
+			return "", errors.Wrap(err, "get primary")
+		}
+	} else {
+		podIPs := make([]string, 0, cluster.Spec.MySQL.Size)
+		for i := 0; i < int(cluster.Spec.MySQL.Size); i++ {
+			podIPs = append(podIPs, mysql.FQDN(cluster, i))
+		}
+		primary, replicas, err = getTopology(podIPs, operatorPass)
+		if err != nil {
+			return "", errors.Wrap(err, "select donor")
+		}
+	}
+	src := ""
+	l.Info("get topology", "primary", primary, "replicas", replicas)
```

Comment:
it'd be better to make this log `DEBUG`
```suggestion
	l.V(1).Info("get topology", "primary", primary, "replicas", replicas)
```

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/157/commits/4c0b10032a5b2cb1d8835be51c25d6d5a8c2d51b